src/effects/effectchainslot.h

```diff
@@ -35,6 +35,7 @@ class EffectChainSlot : public QObject {
 
     void loadEffectChain(EffectChainPointer pEffectChain);
     EffectChainPointer getEffectChain() const;
+    EffectChainPointer getAndEnsureEffectChain(EffectsManager* pEffectsManager);
```

Comment:
I think getOrCreate() is a common naming. That the generated chain is empty should be intuitive and does not need to be mentioned in the function name. Those details are ideally documented separately as a comment to keep the function names short and concise.